dace/frontend/python/replacements.py

```diff
@@ -102,6 +104,64 @@ def _define_streamarray(pv: 'ProgramVisitor',
     return name
 
 
+@oprepo.replaces('numpy.array')
+@oprepo.replaces('dace.array')
+def _define_literal_ex(pv: 'ProgramVisitor',
+                       sdfg: SDFG,
+                       state: SDFGState,
+                       obj: Any,
+                       dtype: dace.typeclass = None,
+                       copy: bool = True,
+                       order: str = 'K',
+                       subok: bool = False,
+                       ndmin: int = 0,
+                       like: Any = None,
+                       storage: Optional[dtypes.StorageType] = None,
+                       lifetime: Optional[dtypes.AllocationLifetime] = None):
+    """ Defines a literal array in a DaCe program. """
+    if like is not None:
+        raise NotImplementedError('"like" argument unsupported for numpy.array')
+
+    name = sdfg.temp_data_name()
+    if dtype is not None and not isinstance(dtype, dtypes.typeclass):
+        dtype = dtypes.typeclass(dtype)
+
+
+    # From existing data descriptor
+    if isinstance(obj, str):
+        desc = dcpy(sdfg.arrays[obj])
+        if dtype is not None: 
+            desc.dtype = dtype
+    else:  # From literal / constant
+        if dtype is None:
+            arr = np.array(obj, copy=copy, order=order, subok=subok, ndmin=ndmin)
+        else:
+            npdtype = dtype.as_numpy_dtype()
+            arr = np.array(obj, npdtype, copy=copy, order=order, subok=subok, ndmin=ndmin)
+        desc = data.create_datadescriptor(arr)
+
+    # Set extra properties
+    desc.transient = True
```

Comment:
So literal arrays are always transient?

Reply:
@TizianoDeMatteis yes, since they are internal, they can always be removed/modified/relayouted/etc.